backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -66,14 +71,27 @@ private async void OnMissionRunCreated(object? sender, MissionRunCreatedEventArg
                 return;
             }
 
+            string missionRunIdToStart = missionRun.Id;
+
+            _scheduleLocalizationSemaphore.WaitOne();
+
+            string? localizationMissionRunId = null;
+            try { localizationMissionRunId = await LocalizationService.EnsureRobotIsCorrectlyLocalized(missionRun.Robot, missionRun); }
+            catch (Exception ex) when (ex is AreaNotFoundException or DeckNotFoundException) { return; }
```

Comment:
Why catch in this manner?

Reply:
Based on the exception, we might want to cancel the mission

Reply:
Changed to cancelling the mission for all exceptions